backend/src/documentation/padding-oracle.md

```diff
@@ -0,0 +1,214 @@
+
+Well, imagine if your plaintext is literally `POTATO\x02\x02`. How could you distinguish a plaintext which content is `POTATO` (6 bytes) and `POTATO\x02\x02` (the `\x02\x02` at the end it seems  padding, but in reality it's part of the string). As a result, a full length block like `POTATOES` (8 bytes or chars) is padded to `POTATOES\x08\x08\x08\x08\x08\x08\x0\x08` which in hexadecimal is `0x504f5441544f45530808080808080808`.
+
+### **Oracle**
+
+An Oracle is a system (e.g. Web Application) that accepts arbitrary ciphertexts and it gives **different responses** where the padding in a CBC-encrypted ciphertext is **valid** or **not**. This system will perform cryptographic operations on behalf of the user or attackers like a black box which returns success or error responses. In other words, a padding oracle takes encrypted data or ciphertext from the user, tries to decrypt it, then responds if the padding was correct or not. The attacker can use this information to decrypt the original message.
+
+
+> Now you have the main knowledge to understand how this attack is performed. Let's get to the point.
+
+
+## The Attack
+
+To perform this attack is not needed to know the key, any plaintext or be an expert in math. Just access to the cipher text and to the decryption method is enough. Firstly, pay attention to the diagram shown below of how the CBC mode decrypts a given ciphertext.
+
+![CBC Decryption](/documentation/img/cbc_decrypt.png "CBC Decryption")
+
+The most relevant point is how each block decryption ends. 
+
+> **The ciphertext passes through the block cipher decryption and then its output is XORed with the previous ciphertext block**. The XOR of these two strings with the correct padding gives the plaintext message.
+
+
+Imagine you have made a random request to a web application you use with the following **encrypted cookie value** and you want to decrypt it to know its content:
+
+> 6A211E23A323D4D54529238A
+
+Assume the application is encrypting this value in CBC mode and PKCS#7 standard. The application receives this value, decrypts it, and sends the response based on it. The perfect scenario for a padding oracle attack! 
+
+With this knowledge, there are 3 possible cases:
+
+1. The ciphertext is valid - a successful response is sent;
+
+2. Invalid ciphertext (with improper padding) -  error message;
+
+3. Valid ciphertext with invalid padding -  error message.
+
+To make this clear, if you can send different ciphertexts and result in different outputs upon its decryption, you know if the padding is valid or not. With this kind of knowledge, the attacker has the ability to decrypt any ciphertext.
+
+To conclude, the goal is to find the correct ciphertext payload to get a successful response (200 OK), meaning we have the right value for the byte we want to decrypt. If it responds with an error (usually 500 error), it means the tampered ciphertext does not decrypt to a valid message. We can try other payloads to find a proper match.
```

Comment:
I wouldn't use the status codes here as they are not the whole truth... you may have an error with a 200 for example... really depends on the implementation